compression.go

```diff
@@ -81,7 +81,7 @@ func (c *compressionPool) Decompress(dst *bytes.Buffer, src *bytes.Buffer, readM
 		return errorf(CodeInvalidArgument, "get decompressor: %w", err)
 	}
 	reader := io.Reader(decompressor)
-	if readMaxBytes > 0 && readMaxBytes < math.MaxInt64 {
+	if readMaxBytes > 0 && int64(readMaxBytes) < math.MaxInt64 {
```

Comment:
For 32 bit the clause after the `&&` will always be true.
Why not `readMaxBytes < math.MaxInt`?

Reply:
The main reason this exists is just not to overflow when adding one below. Also https://pkg.go.dev/io#LimitReader and https://pkg.go.dev/io#ReaderFrom both deal with `int64` so easier to use that consistently.

Reply:
But if `readMaxBytes` is `math.MaxInt32`, and you add anything to it, it will overflow?
If everything else is `int64` then the most correct thing to do is to use `int64` consistently everywhere.
If that isn't possible, then the bounds check should be checking against the variable's int size.

Reply:
I'd either change the type of `readMaxBytes` to be `int64`, or create a local `int64` set to `int64(readMaxBytes)` and use that for the remainder of the function. Reduces the risk of missing inserting a cast somewhere, or having to constantly reason about the type and decide whether to cast.

Reply:
Good call out - opened https://github.com/bufbuild/connect-go/pull/319 to improve.